src/entities/join/types.ts

```diff
@@ -54,6 +54,7 @@ export type JoinQueryResult = {
     amountsIn: TokenAmount[];
     fromInternalBalance: boolean;
     tokenInIndex?: number;
+    bptIndex?:number;
```

Comment:
Lets create a `BaseJoinQuery` type and then separate query types for each pool type, e.g. `WeightedQueryResult`, `StableQueryResult`. Then `JoinQueryResult` can be a union of those. That way we know what properties belong to what pool type. e.g. we know `bptIndex` belongs to stable pools.